helpers/preview/bin/preview-rst

```diff
@@ -0,0 +1,313 @@
+#   new browser window.
+#
+#   This strategy has many advantages:
+#
+#   - At the moment, the Read The Docs theme is used. This styles your RST in
+#     an easy-to-ready manner.
+#
+#   - The `sphinx-autobuild` program features live reloading, meaning that when
+#     you edit the RST file, the HTML will be rebuilt, and the browser window
+#     will automatically refresh to show your changes.
+
+
+# Caution
+# -----------------------------------------------------------------------------
+
+# This script is a work in progress and features may be added, removed, or
+# changed significantly.
+#
+# TODO:
+#
+# - Assorted `TODO` markers scattered throughout the code
+#
+# - Add wrappers for specific editors (e.g., BBEdit)
+#
+# - Needs to exclude some directives (e.g., admonitions) when generating
+#   standalone files to mimic GitHub's limited RST support
+
+
+# Setup
+# =============================================================================
+
+PY='python3.7'
+
+script_file="${0}"
+script_dir=`dirname ${script_file}`
+script_basename=`basename ${script_file}`
+etc_dir="${script_dir}/../etc"
+
+if test -z "${1}"; then
+    echo "Usage: ${script_basename} <RST_FILE>"
+    exit 1
+fi
+
+rst_file=`realpath "${1}"`
+
+if ! test -f "${rst_file}"; then
+    echo "No such file: ${1}"
+    exit 1
+fi
+
+
+# Utility functions
+# =============================================================================
+
+function get_output_dir {
+    input_file="${1}"
+    # We need the following variables so that we can recreate the full path
+    # to the file under the `/tmp` directory.
+    #
+    # Doing it this way ensures that the URL for the output HTML remains the
+    # same no matter how many times you regenerate the HTML. Doing it this way,
+    # instead of creating a random directory name (e.g., using `mktemp`), means
+    # that refreshing the browser works to pick up changes if the file is
+    # modified.
+    input_dirname=`dirname "${input_file}"`
+    output_dir_subpath=`realpath "${input_dirname}"`
+    # Namespace our temporary directories
+    output_dir="/tmp/${script_basename}/output${output_dir_subpath}"
+    # This path should remain constant for subsequent HTML generations of the
+    # same source file path (see above). We leave deletion of this temporary
+    # directory up to the user or the operating system.
+    mkdir -p "${output_dir}"
+    echo "${output_dir}"
+}
+
+function progs_exist {
+    progs="${1}"
+    return_code=0
+    for prog in "${progs}"; do
+        if test -z `which "${prog}"`; then
+            return_code=1
+            echo "The `${prog}` program is not on your PATH." >&2
+        fi
+    done
+    return $return_code
+}
+
+
+# This function creates a temporary Python virtual environment that is shared
+# by every invokation of this script
+function get_venv_dir {
+    tmp_dir="${1}"
+    # Namespace our temporary directories
+    venv_dir="/tmp/${script_basename}/deps/${PY}/venv"
```

Comment:
On many Linux machines, `/tmp` is mounted with the `noexec` flag these days. So, I suggest to use `/var/tmp` instead. On the other hand, for example on my macOS machine, the environment already contains a `TMPDIR` entry.

Maybe define the designated temporary directory at the beginning of the program like:

```sh
TMPDIR="${TMPDIR:-/var/tmp}"
```

and then use `${TMPDIR}` all over the place.

Reply:
came up with a solution using `mktemp`